FirebaseDatabase/Sources/Api/FIRDatabaseQuery.m

```diff
@@ -189,12 +190,37 @@ - (FIRDatabaseQuery *)queryStartingAtValue:(id)startValue
                           priorityMethod:NO];
 }
 
+- (FIRDatabaseQuery *)queryStartingAfterValue:(id)startAfterValue {
+    return [self queryStartingAfterValue:startAfterValue childKey:nil];
+}
+
+- (FIRDatabaseQuery *)queryStartingAfterValue:(id)startAfterValue
+                                     childKey:(NSString *)childKey {
+    if ([self.queryParams.index isEqual:[FKeyIndex keyIndex]]) {
+        @throw [[NSException alloc]
+            initWithName:INVALID_QUERY_PARAM_ERROR
+                  reason:@"You must use queryStartingAtValue: instead of "
+                         @"queryStartingAtValue:childKey: when using "
+                         @"queryOrderedByKey:"
+                userInfo:nil];
+    }
+    if (childKey == nil) {
+        childKey = [FUtilities maxName];
+    } else {
+        childKey = [FNextPushId nextAfter:childKey];
+    }
+    return [self queryStartingAtInternal:startAfterValue
+                                childKey:childKey
+                                    from:@"queryStartingAfterValue:childKey:"
+                          priorityMethod:NO];
+}
+
 - (FIRDatabaseQuery *)queryStartingAtInternal:(id<FNode>)startValue
                                      childKey:(NSString *)childKey
                                          from:(NSString *)methodName
                                priorityMethod:(BOOL)priorityMethod {
     [self validateIndexValueType:startValue fromMethod:methodName];
-    if (childKey != nil) {
+    if (childKey != nil && ![childKey isEqual:[FUtilities maxName]]) {
```

Comment:
This now allows `maxName` to be passed in startAt. We should probably move the validation to each callsite.